### ISSUE TITLE
Add Source Code

### ISSUE THREAD

**Issue by babeuh** [openspeedtest/Speed-Test#26]
Please add the source code somewhere in this repo, it has already been shared on reddit with the pastebin link. This would make it available to more users.

I understand that you want to refactor your code before making it public but it would better to at least add that pastebin link somewhere before that (though adding the file would be better)

btw, deleting issues on github makes you look shady, just close them without deleting if you don’t agree with the issues

**Comment by openspeedtest** [comment]
You can take the js file posted here.

minified version, that means removed comments and reduced file size using google closure compiler. it will decrease file size and increase code performance.

Unminified Code : - https://pastebin.com/4J59Hud2

Only 750 lines of code. you can review it manually, each line by line, if you need. I have not obfuscated the code.

Obfuscated Code Looks like this :- https://pastebin.com/53SX41EK

Use any JavaScript unminifier view it. 

It is client side javascript. 

It's not like c++ compailed binary. 

Anyone can view it also see how it works through chrome browser console.

**Comment by openspeedtest** [comment]
I think you got confused. I posted the same code i posted here, it's unminifed using unminify.com, with that you see almost 90% of the code just like original, 10% will be little difficult to understand, but if you have good javascript skills you can do whatever you want with it. Now with MIT licence anyone can change colours or edit html page to match there design or add some custom behaviour without any issues. That is what this project can do now.

**Comment by openspeedtest** [comment]
I spent countless hours day and night throughout these years to make it what it is;[ I didn't get any help to solve complex challenges](https://github.com/openw3/Open-Source-Internet-Speed-Test--Broadband-bandwidth-test/blob/master/LICENSE.md) from the community. Still, I love to give something useful for Free. So that is my GitHub project. One or more people (Probably A Company that is profiting from OpenSource Projects like to rebrand my work or integrate it in to its proprietary software.) who is posted bad comments on Reddit. So I am not obligated to anyone. I don't want to give years of my work to people like this. Even if i decided to give it away, i need to spend another 100 or more hours to make it good for a public project. _For complete refactoring, it needs time, and it will also introduce new bugs. Currently, it's not a priority for me_

**Comment by openspeedtest** [comment]
I have a very bad habit, I like to answer all questions till the last one. That is very bad if you run OpenSource Project. People will keep asking for new stuff and need to support existing stuff almost like a day job. If this effort makes money like GitLab or Portainer. I will put one or two employees to support the project. now I am learning new stuff and trying to get a job. Strictly I don't have time to support this project, One day I will post the complete code with comments. But I don't know when....

**Comment by PedroHase** [comment]
Dear @openspeedtest I believe there are some misunderstandings here. Please allow me to clear these up.

### Unminified code !== source code

Maybe for you it's easy to understand your unminified code because you know how it works, but for outsiders minified code is terrible for many reasons:
- Variables are being renamed (a,b,c,d,...) making it impossible to know what kind of assignment happened
- Code is being restructured & shorted and makes it very hard to understand the logic & flow of the program

Just as an example, look at following code. 

```js
r.prototype.LiveSpeed = function (a, c) {
        "countDown" === c
            ? ((c = a.toFixed(0)), (this.oDoLiveSpeed.el.textContent = c))
            : "speedToZero" === c
            ? ((c = a.toFixed(1)), (this.oDoLiveSpeed.el.textContent = c), (this.oDoTopSpeed.el.textContent = "1000+"), (this.oDoTopSpeed.el.style.fontSize = 16.9), (this.oDoTopSpeed.el.style.fill = "gray"))
            : "Ping" === c
            ? 1 <= a && 1e4 > a
                ? (this.oDoLiveSpeed.el.textContent = Math.floor(a))
                : 0 <= a && 1 > a && (this.oDoLiveSpeed.el.textContent = a)
            : (0 == a && ((c = a.toFixed(1)), (this.oDoLiveSpeed.el.textContent = c)),
              1 >= a && 0 < a && ((c = a.toFixed(3)), (this.oDoLiveSpeed.el.textContent = c)),
              1 < a && ((c = a.toFixed(1)), (this.oDoLiveSpeed.el.textContent = c)),
              1e3 >= a && ((this.oDoTopSpeed.el.textContent = "1000+"), (this.oDoTopSpeed.el.style.fontSize = 16.9), (this.oDoTopSpeed.el.style.fill = "gray")),
              1010 <= a && ((this.oDoTopSpeed.el.textContent = 1e3 * Math.floor(a / 1010) + "+"), (this.oDoTopSpeed.el.style.fill = "#434040"), (this.oDoTopSpeed.el.style.fontSize = "20px")));
    };
```

You might think it's easy to understand because you know it's purpose, but for us we really can only guess because the minifier introduces "noisy" expressions like `? 1 <= a && 1e4 > a` or `1 >= a && 0 < a` which makes it really difficult to understand what happens. Also most of us do not want or have the time to spend hours just trying to decode the minified code.  And yes, even if it's "just" 750 lines, since they're unminified (i.e. not the source), it will take very long decode for the average dev since it is very uncommon for having to decipher unminified code.

### The "old" version

I do not understand why you (randomly?) keep pointing to [this old repo](https://github.com/openw3/Open-Source-Internet-Speed-Test--Broadband-bandwidth-test/blob/master/LICENSE.md). The repo never had any contents except for this license file (or maybe you removed the commits?). Whatever old "version" of the source code there was, it is gone (or never was there) and we cannot access it.

### Code being "stolen"

Yes it can be frustrating if companies are using open-source code/projects (or rebranding it as their own work)  without compensation, or at least crediting the original project / author. But there are a few ways to actually work around that, e.g. through a specific license (e.g. AGPL, Apache) or by asking for sponsorship / crediting you, otherwise you shouldn't have published your work, because that is how open-source works, unfortunately,

### Regarding code quality and the necessity to clean up

It seems that you are quite self-conscious about your code, and there is nothing wrong about that but there also is nothing to fear if you let outsiders take a look. Quite the opposite, external collaborators could actually help cleaning up the code and making it better than ever, or point out flaws and help you improve. Or you could work with some people together and work on a completely new version or let the community do the refactoring, etc. That's why open-source projects exist and the source code is open, so that everyone can take a look and suggest or make improvements.

### The importance of publishing the source code

[I tried to explain this point on Reddit about two months ago.](https://reddit.com/r/javascript/comments/vtbah8/_/if6pos2/?context=1). TL;DR: Not publishing the source code is a big red flag and undermines the credibility of this great project because people may think that you may hide something and it is in stark contrast to the `openspeedtest` name of the project, since not publishing the source code is equal to keeping it closed code. 

### To summarize

I hope you understand that there is no valid reason for you not to publish the source code. While you claim that you do not want to publish the code because you want to comment and refactor the code (which I believe everyone would highly appreciate), I don't see them as a strong enough argument to hold back the source code, since this does more harm than good for the project's reputation and goes against the idea of open-source.

I also can understand that you feel uncomfortable releasing the code because you feel it is "just not there", but I believe since you already do know that you could use the power of open-source to work with collaborators to greatly improve the project when you release the source code :)

**Comment by openspeedtest** [comment]
The project is now almost ten years old.

Last year posted this on GitHub. Previously, I provided this as a download from my website for custom implementation. Now, it's on GitHub. that is the difference.

The main reason I posted it on GitHub is for CI (continuous integration)

I am learning CI/CD.

The main reason I didn't post the committed code is because of this.

I used "this.variable" It can be "var variable", "let variable", "const variable"

All the above code works, but people will post arguments about it.

For example, why do you use "this.var"? Just use "let.var" and so on.

When I posted in StackOverflow  I got comments like this.

Then another issue is " This is not MVC ". So you cannot call it MVC.

Hey, no matter what, it's working for me.

Next, why do you use inheritance when you don't need this in this function?

You know people will bring 1000 + questions like this.

Some of them are valid. I need to learn more about JavaScript and correct those mistakes.

That is the main reason I am afraid to post this.

I am not ready (Knowledge in Javascript) to handle this many questions. So maybe I need to learn javascript again and make it somewhat ok. that is why I posted I need more time.

I am learning Python and will work on a CLI client for openSpeedTest.

Then I will come back to JavaScript and do the refactoring. But I am the only one here to do this work. So I will make it one day.

**Comment by PedroHase** [comment]
> When I posted in StackOverflow I got comments like this.

Many people on SO (StackOverflow) are somewhat elitist and there are [countless](https://preview.redd.it/xeqte6t1sou71.png?width=640&crop=smart&auto=webp&s=8f0618c8d8269d8cd0393b39b76660162f24dd6e) [memes](https://c.tenor.com/lYJUdIQn1IEAAAAC/stackoverflow-programming.gif) about that. However, most of the time they do give good advice, but this does involve some nitpicking of course which in turn gives a good opportunity to learn and reflect.

> You know people will bring 1000 + questions like this.

Github is **NOT** SO, I can guarantee you this won't happen. 99% of the time people won't nitpick every line of code, but rather just ask for new features or problems. And even if someone would nitpick the code, you could just simply ask them to create a PR (Pull Request) which you could then merge to improve the code. And, if you make it clear that you do not have great knowledge of JS (e.g. explaining it in the `README`), people will understand. 

Again, I can understand your concerns, most developers go through this (also called "imposter syndrome"), but I can only urge you to publish the source code, otherwise people may wonder where the `Open` in Openspeedtest is if the most important part is actually closed-source.

**Comment by openspeedtest** [comment]
Anyway, I will work on the CLI client. It will be in the MIT license. Let's see how people contribute or help me on CLI. I have zero knowledge of Python. If I did receive good support, I would make Everything fully OpenSource. But another question is, who cares? Now we have many speed tests. It's in oversupply. Already my project is irrelevant. So I am moving to something else. Now it's not worth my 100+ Hours into this to make the commented code out. Directly you can change colours without touching the JS. That is the scope of the project right now.

**Comment by openspeedtest** [comment]
I will delete this issue once I post the unminifed code because this thread will become irrelevant soon. 
Now learning Python and Golang. 
-> Get the CLI client ready. 
-> Work on the Legacy JS and remove offensive comments, Maybe all comments. 
-> Add more descriptive Variable names. 
I will do this by the end of this year. 
^^ is just a roadmap that may or may not happen. 
I will wait till Dec 31 2022

**Comment by JocelynDelalande** [comment]
A big thanks for going towards opensourcing OpenSpeedTest !

This software definitely has an interest and seems quite the only in its category (open-source HTML speedtest, fully static).

I can understand that you want a « clean » source code before releasing it (especially removing offensive comments :-s), something you can be proud of. But please don't put too much stress on you about having a « perfect » software.

Community members have expressed interest into having the source code, not into having a clean code :).

 To give my standpoint, I'd love to use this repo and the docker image, but cannot trust it enough to deploy it since readable source code of both webapp and dockerfile are published.

Happy by advance to see progress near the end of the year then :-).

**Comment by openspeedtest** [comment]
here is the docker file.
````
# By OpenSpeedTest
# Dockerfile for https://hub.docker.com/r/openspeedtest/latest
FROM nginxinc/nginx-unprivileged:stable-alpine

LABEL maintainer "OpenSpeedTest.com <support@OpenSpeedTest.com>"

COPY /files/OpenSpeedTest-Server.conf /etc/nginx/conf.d/OpenSpeedTest-Server.conf
RUN rm /etc/nginx/nginx.conf
COPY /files/nginx.conf /etc/nginx/
COPY /files/www/ /usr/share/nginx/html/
COPY /files/nginx.crt /etc/ssl/
COPY /files/nginx.key /etc/ssl/



USER root

RUN rm -rf /etc/nginx/conf.d/default.conf \
	&& chown -R nginx /usr/share/nginx/html/ \
	&& chmod 755 /usr/share/nginx/html/downloading \
	&& chmod 755 /usr/share/nginx/html/upload \
	&& chown nginx /etc/nginx/conf.d/OpenSpeedTest-Server.conf \
	&& chmod 400 /etc/nginx/conf.d/OpenSpeedTest-Server.conf \
	&& chown nginx /etc/nginx/nginx.conf \
	&& chmod 400 /etc/nginx/nginx.conf 

RUN apk update && apk add ca-certificates && rm -rf /var/cache/apk/* 
RUN update-ca-certificates

USER 101

EXPOSE 3000 3001

CMD ["nginx", "-g", "daemon off;"]

````
Now I am trying to learn Python and GoLang to write a CLI-Client for OpenSpeedTest.
I think you know Python (by looking at your GitHub Profile) I am stuck with this problem https://github.com/aio-libs/aiohttp/issues/6912 Can you help me to fix this issue? 


Security and Performance is my primary concern which is why we have only "STATIC FILES"
Min Js will make requests to your server and test the connection speed. also, do some animation that is it.

If you are paranoid and don't know how to isolate a service, wait for the source code. 

Many ISPs, Public and private organizations use OpenSpeedTest. 

https://cll-east-speedtest.cisco.com/
http://speedtest.sgix.sg/
https://speedtest.pnl.gov/
https://speedtest2.pnl.gov/

**Comment by openspeedtest** [comment]
Verify Source code?
Change app-2.5.3.min.js to app-2.5.3.js
Goodbye
